src/Upgrader/Business/PackageManager/Client/Composer/ComposerClient.php

```diff
@@ -8,29 +8,118 @@
+     */
+    protected $composerJsonReader;
+
+    /**
+     * @var \Upgrader\Business\PackageManager\Client\Composer\Lock\Reader\ComposerLockReaderInterface
+     */
+    protected $composerLockReader;
+
     /**
      * @param \Upgrader\Business\Command\CommandInterface $composerUpdateCommand
+     * @param \Upgrader\Business\PackageManager\Client\Composer\Command\ComposerRequireCommandInterface $composerRequireCommand
+     * @param \Upgrader\Business\PackageManager\Client\Composer\Json\Reader\ComposerJsonReaderInterface $composerJsonReader
+     * @param \Upgrader\Business\PackageManager\Client\Composer\Lock\Reader\ComposerLockReaderInterface $composerLockReader
      */
-    public function __construct(CommandInterface $composerUpdateCommand)
-    {
+    public function __construct(
+        CommandInterface $composerUpdateCommand,
+        ComposerRequireCommandInterface $composerRequireCommand,
+        ComposerJsonReaderInterface $composerJsonReader,
+        ComposerLockReaderInterface $composerLockReader
+    ) {
         $this->composerUpdateCommand = $composerUpdateCommand;
+        $this->composerRequireCommand = $composerRequireCommand;
+        $this->composerJsonReader = $composerJsonReader;
+        $this->composerLockReader = $composerLockReader;
     }
 
     /**
-     * @return \Upgrader\Business\Command\ResultOutput\CommandResultOutput
+     * @return \Upgrader\Business\Command\Response\CommandResponse
      */
-    public function runUpdate(): CommandResultOutput
+    public function runUpdate(): CommandResponse
     {
         return $this->composerUpdateCommand->run();
     }
+
+    /**
+     * @return string
+     */
+    public function getProjectName(): string
+    {
+        $composerJsonContent = $this->composerJsonReader->read();
+
+        return $composerJsonContent[self::NAME_KEY];
+    }
+
+    /**
+     * @return array
+     */
+    public function getComposerJsonFile(): array
+    {
+        return $this->composerJsonReader->read();
+    }
+
+    /**
+     * @return array
+     */
+    public function getComposerLockFile(): array
+    {
+        return $this->composerLockReader->read();
+    }
+
+    /**
+     * @param \Upgrader\Business\PackageManager\Entity\Collection\PackageCollection $packageCollection
+     *
+     * @return \Upgrader\Business\Command\Response\CommandResponse
+     */
+    public function require(PackageCollection $packageCollection): CommandResponse
+    {
+        $this->composerRequireCommand->setPackageCollection($packageCollection);
+
+        return $this->composerRequireCommand->run();
+    }
+
+    /**
+     * @param string $packageName
+     *
+     * @return string|null
+     */
+    public function getPackageVersion(string $packageName): ?string
+    {
+        $composerLock = $this->composerLockReader->read();
+
+        foreach ($composerLock[self::PACKAGES_KEY] as $package) {
```

Comment:
Is it a part of the composer lock business model?

Reply:
Yes